### PR TITLE
clean filenames to adhere standard

### PR DESCRIPTION
**Pull request by s-l-teichmann**
Only allow uploads of validated filenames.

### CODE REVIEW

**Comment by s-l-teichmann** [comment]
Addresses issue #88

File: util/file.go [hunk 1]
```diff
@@ -19,15 +19,19 @@ import (
...
-	stripSlashes  = strings.NewReplacer(`/`, ``, `\`, ``)
-)
+var invalidRune = regexp.MustCompile(`([^a-z0-9\+_-])`)
 
-// CleanFileName removes the "/" "\" charachters and replace the two or more
-// occurences of "." with only one from the passed string.
+// CleanFileName replaces invalid runes with an underscore.
+// If the filename does not end with '.json' it will be appended.
+// The filename is converted to lower case.
+// https://docs.oasis-open.org/csaf/csaf/v2.0/cs01/csaf-v2.0-cs01.html#51-filename
```
Comment (bernhardreiter) [inline util/file.go:27]: Should habe been https://docs.oasis-open.org/csaf/csaf/v2.0/csd02/csaf-v2.0-csd02.html#51-filename ?

**Review by bernhardreiter — CHANGES_REQUESTED** [review]
Should be changed to follow the updated rules from https://docs.oasis-open.org/csaf/csaf/v2.0/csd02/csaf-v2.0-csd02.html#51-filename

File: util/file_test.go [hunk 2]
```diff
@@ -5,6 +5,22 @@ import (
...
 )
 
+func TestCleanFileName(t *testing.T) {
+	for _, x := range [][2]string{
+		{`HELLO`, `hello.json`},
+		{`hello`, `hello.json`},
+		{`cisco-sa-20190513-secureboot.json`, `cisco-sa-20190513-secureboot.json`},
+		{``, `.json`},
+		{`..`, `__.json`},
+		{`../..`, `_____.json`},
```
Comment (bernhardreiter) [inline util/file_test.go:15]: several underscores shall be grouped to one (in the csd02 specification).

**Comment by bernhardreiter** [comment]
We would need a function to test filenames, if they are compliant, so we can reject them (as required to solve #88

**Review by bernhardreiter — APPROVED** [review]

**Pull request merged**